packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -100,31 +99,37 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
 
         // Remove tags until there is space for the collapse button and action button (if present) on the last row.
         let buttonsWidth = buttons.reduce((acc, curr) => acc += curr.getBoundingClientRect().width, 0);
+        buttonsWidth += parseInt(window.getComputedStyle(buttons[buttons.length - 1]).marginRight, 10) * 2;
         let end = direction === 'ltr' ? 'right' : 'left';
-        let containerEnd = currContainerRef.getBoundingClientRect()[end];
+        let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
         let lastTagEnd = tags[index - 1]?.getBoundingClientRect()[end];
+        lastTagEnd += parseInt(window.getComputedStyle(tags[index - 1]).marginRight, 10);
         let availableWidth = containerEnd - lastTagEnd;
-        for (let tagWidth of tagWidths.reverse()) {
-          if (availableWidth >= buttonsWidth || index <= 1 || index >= state.collection.size) {
-            break;
-          }
-          availableWidth += tagWidth;
+
+        while (availableWidth < buttonsWidth && index < state.collection.size && index > 0) {
+          availableWidth += tagWidths.pop();
           index--;
         }
-        return {visibleTagCount: index, showCollapseButton: index < state.collection.size};
+        let tagStyle = window.getComputedStyle(tags[0]);
+        let maxHeight = (parseInt(tagStyle.height, 10) + parseInt(tagStyle.marginTop, 10) * 2) * maxRows;
+        return {
+          visibleTagCount: index,
+          showCollapseButton: index < state.collection.size,
+          maxHeight
+        };
```

Comment:
I was able to get it the taggroup into the following state where the "clear" button wasn't visible:
<img width="577" alt="image" src="https://user-images.githubusercontent.com/9661127/226753098-607a9697-147f-4918-84c0-fbbc590885fb.png">
This was done via this [story](https://reactspectrum.blob.core.windows.net/reactspectrum/1506856214b1e7435876cd11fb62d193801ea221/storybook/index.html?path=/story/taggroup--with-action-and-max-rows&args=maxRows:3&providerSwitcher-express=false) by setting maxRows to 3. Do we need to take that button's existence into account as well?

Reply:
Good catch. We shouldn't be setting the maxHeight in this case. I'll update the logic to reflect that.